src/lib/hash/blake2/blake2b.h

```diff
@@ -17,27 +18,41 @@ namespace Botan {
 /**
 * BLAKE2B
 */
-class BLAKE2b final : public HashFunction
+class BLAKE2b final : public HashFunction, public SymmetricAlgorithm
    {
    public:
       /**
       * @param output_bits the output size of BLAKE2b in bits
       */
       explicit BLAKE2b(size_t output_bits = 512);
+      BLAKE2b(const SymmetricKey& key, size_t output_bits = 512);
+      BLAKE2b(const uint8_t key[], size_t key_length, size_t output_bits = 512);
+
+      template<typename Alloc>
+      BLAKE2b(const std::vector<uint8_t, Alloc>& key, size_t output_bits = 512) :
+          BLAKE2b(key.data(), key.size(), output_bits)
+      {
+      }
 
       size_t hash_block_size() const override { return 128; }
       size_t output_length() const override { return m_output_bits / 8; }
+      size_t key_size() const { return m_key_size; }
+
+      Key_Length_Specification key_spec() const override;
 
       HashFunction* clone() const override;
       std::string name() const override;
       void clear() override;
 
       std::unique_ptr<HashFunction> copy_state() const override;
 
-   private:
+   public:
```

Comment:
`key_schedule` as well as `add_data` and `final_result` should be `private` - they are only called via other functions defined on the base class.

[This is mostly harmless now that the algorithm types aren't publicaly exposed but I'd like to keep it clear what is or is not callable APIs]